main.cpp

```diff
@@ -108,7 +108,410 @@ write 3 UDTs below that EACH have:
  */
 
 #include <iostream>
+#include <vector>
+#include <string>
+#include <unordered_set>
+
+struct Loan
+{
+    double loanAmount;
+    double interestRateInPercent;
+    unsigned long numPayments;
+    bool isSecured;
+    std::vector<double> payments;
+
+    Loan(double amount, double rate, unsigned long numPay, bool secured);
+    ~Loan();
+    void calculatePayments();
+    double calculateOverpayment();
+};
+//--
+Loan::Loan(double amount, double rate, unsigned long numPay, bool secured):
+    loanAmount(amount), interestRateInPercent(rate), numPayments(numPay), isSecured(secured)
+{
+    if(isSecured)
+    {
+        interestRateInPercent *= 0.9;
+    }
+    calculatePayments();
+}
+//--
+Loan::~Loan()
+{
+    std::cout << "Loan destructor called!" << std::endl;
+}
+//--
+void Loan::calculatePayments()
+{
+    payments.clear();
+    for(unsigned int i = 0; i < numPayments; i++)
+        {
+            double currentLoanAmount = loanAmount;
+            double payment = currentLoanAmount * (interestRateInPercent / 12 / 100.0) + loanAmount / static_cast<double>(numPayments);
+            currentLoanAmount -= loanAmount / static_cast<double>(numPayments);
+            payments.push_back(payment);
+        }
+}
+//--
+double Loan::calculateOverpayment()
+{
+    double amountToPay = 0.0;
+    for(unsigned int i = 0; i < numPayments; i++)
+        {
+            amountToPay += payments[i];
+        }
+    return amountToPay - loanAmount;
+}
+//--
+struct Gym 
+{
+    struct Member 
+    {
+        std::string name;
+        static int memberId;
+        int age;
+        bool isActive = true;
+        std::vector<std::string> activityLog;
+
+        Member(std::string n, int a);
+        ~Member();
+        void logActivity(const std::string& activity);
```

Comment:
const and references have not been introduced yet.
please refrain from using language features that haven't been introduced in the course yet. 
- [ ] revise